src/bio/bio_wal.c

```diff
@@ -796,10 +770,13 @@ bio_wal_commit(struct bio_meta_context *mc, uint64_t tx_id, struct umem_action *
 	unsigned int		 blks, unused_off;
 	unsigned int		 tot_blks = si->si_header.wh_tot_blks;
 	unsigned int		 blk_bytes = si->si_header.wh_blk_bytes;
+	uint64_t		 tx_id = tx->utx_id;
 	int			 iov_nr, rc;
 
 	/* Calculate the required log blocks for this transaction */
-	calc_trans_blks(umem_act_nr(actv), umem_act_payload_sz(actv), blk_bytes, &blk_desc);
+	calc_trans_blks(umem_tx_act_nr(mc->mc_meta->bic_umem, tx),
```

Comment:
Minor: This 'bic_umem' was used for bio io context, it could be changed in later BIO code cleanup. If umem_tx_act_xxx() function requires umem_instance, can we just put the umem_instance in umem_tx (maybe replace the 'utx_class' with 'umem_instance')?

Reply:
seems fine to me, can change it later when add more detailed code.

Reply:
agree